### ISSUE TITLE
Box template with a box with children

### ISSUE THREAD

**Issue by mtareiling** [OFFLINE-GmbH/oc-boxes-support#79]
Is it possible to create a template with a box that has some standard children boxes?
Almost like a template in a template....
example:

```
-   name: columns
    handle: columns
    contexts:
    - page
    boxes:
    - partial: header
      locked: ['position', 'deletion']
    - partial: column-2
    data: header
```

**Comment by mtareiling** [comment]
Solved this by adding support for `children` in file `\OFFLINE\Boxes\Classes\Traits\HasBoxes`:

```
    /**
     * Apply a Boxes template configuration.
     */
    public function applyBoxesTemplate(string $handle)
    {
        $templates = PartialReader::instance()->getBoxesConfig()->get('templates');

        if ($templates->has($handle) === false) {
            throw new LogicException("[CMS.Boxes] The template '{$handle}' does not exist.");
        }

        $boxes = $templates->get($handle)->get('boxes');

        $this->boxes()->saveMany($this->createBoxesFromConfig($boxes));
    }

    /**
     * Recursively create Boxes from a configuration array.
     */
    protected function createBoxesFromConfig(Collection $boxes, $parentBoxId = null): Collection
    {
        $configuredBoxes = collect([]);

        foreach ($boxes as $box) {
            if ($parentBoxId !== null) {
                $box['parent_id'] = $parentBoxId;
            }

            $mainBox = Box::make($box);
            $configuredBoxes->push($mainBox);

            if (isset($box['children'])) {
                $mainBox->save();
                $configuredBoxes = $configuredBoxes->merge($this->createBoxesFromConfig(collect($box['children']), $mainBox->id));
            }
        }

        return $configuredBoxes;
    }
```

**Comment by tobias-kuendig** [comment]
Thank you very much for the implementation @mtareiling! I have included this addition in v3.1.5.
